### PR TITLE
HPCC-29904 Remove generated code that builds the XSD and WSDL

### PR DESCRIPTION
**Pull request by kenrowland**
Remove code from HIDL that generates the code that generates the XSD and
WSDL forms.

Signed-Off-By: Kenneth Rowland kenneth.rowland@lexisnexisrisk.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/bindings/http/platform/httpbinding.hpp [hunk 1]
```diff
@@ -24,6 +24,8 @@
 
 #include "bindutil.hpp"
 #include "seclib.hpp"
+//#include "esdl_def_helper.hpp"
```
Comment (afishbeckln) [inline esp/bindings/http/platform/httpbinding.hpp:27]: Just remove commented code.  The history is still there.

**Review by afishbeckln — COMMENTED** [review]
@kenrowland It's great to see large chunks of the generated code being replaced with shared code.  Just want to make sure thought... this has been well tested to make sure no functionailty is missing?

**Review by afishbeckln — APPROVED** [review]
There are a few places where commented code should be removed, but approving as is.  You can still remove those comments.

File: tools/hidl/hidlcomp.cpp [hunk 2]
```diff
@@ -3504,224 +2900,12 @@ void EspMessageInfo::write_esp()
...
-            outf("\t{\n");
-            outf("\t\tadded.setValue(\"%s\",1);\n", (*it).first.c_str());
-            outf("\t\tC%s%s::getMapInfo(info,added);\n",((*it).second & MIT_IsEnum)?"X":"",(*it).first.c_str());
-            outf("\t}\n");
-        }
-    }
-
     outs("}\n");
 
     //=======================================================================================
```
Comment (asselitx) [inline tools/hidl/hidlcomp.cpp:2905]: I think this  three line comment block is for the getHtmlForm function output which was removed, so the comment should be too.

File: tools/hidl/hidlcomp.cpp [hunk 3]
```diff
@@ -4995,11 +3951,8 @@ void EspServInfo::write_esp_binding_ipp()
     //method ==> processRequest
     outs("\tvirtual int processRequest(IRpcMessage* rpc_call, IRpcMessage* rpc_response);\n");
 
-    //method ==> getXsdDefinition
-    outs("\tint getXsdDefinition(IEspContext &context, CHttpRequest* request, StringBuffer &content, const char *service, const char *method, bool mda);\n");
-
-    //method ==> getMethodHtmlForm
-    outs("\tvirtual int getMethodHtmlForm(IEspContext &context, CHttpRequest* request, const char *serv, const char *method, StringBuffer &page, bool bIncludeFormTag);\n");
+    // method ===> getXmlFilename
```
Comment (asselitx) [inline tools/hidl/hidlcomp.cpp:3954]: Correct comment to read getServiceXmlFilename

**Review by asselitx — CHANGES_REQUESTED** [review]
Just a couple small corrections to comments, otherwise looks good.

File: tools/hidl/hidlcomp.cpp (hunk 2 above)
Comment (kenrowland) [reply tools/hidl/hidlcomp.cpp]: Removed.

File: tools/hidl/hidlcomp.cpp (hunk 3 above)
Comment (kenrowland) [reply tools/hidl/hidlcomp.cpp]: Updated.

**Review by asselitx — APPROVED** [review]

**Comment by kenrowland** [comment]
@ghalliday Please merge

**Pull request merged**